src/ucp/tag/eager_snd.c

```diff
@@ -48,30 +60,30 @@ static size_t ucp_tag_pack_eager_dt(void *dest, void *arg)
     ucp_request_t *req = arg;
     size_t length;
 
-    length         = req->send.length;
     hdr->super.tag = req->send.tag;
 
     ucs_assert(req->send.state.offset == 0);
-    ucp_tag_pack_eager_dt_copy(hdr + 1, req->send.buffer, &req->send.state,
-                               length, req->send.datatype);
+    length = ucp_tag_pack_eager_dt_copy(hdr + 1, req->send.buffer,
+                                        &req->send.state, req->send.length,
+                                        req->send.datatype);
+    ucs_assert(length == req->send.length);
     return sizeof(*hdr) + length;
 }
 
 static size_t ucp_tag_pack_eager_sync_dt(void *dest, void *arg)
 {
     ucp_eager_sync_hdr_t *hdr = dest;
     ucp_request_t *req = arg;
-    size_t length;
 
-    length               = req->send.length;
     hdr->super.super.tag = req->send.tag;
     hdr->req.sender_uuid = req->send.ep->worker->uuid;
     hdr->req.reqptr      = (uintptr_t)req;
 
     ucs_assert(req->send.state.offset == 0);
-    ucp_tag_pack_eager_dt_copy(hdr + 1, req->send.buffer, &req->send.state,
-                               length, req->send.datatype);
-    return sizeof(*hdr) + length;
+    return sizeof(*hdr) + ucp_tag_pack_eager_dt_copy(hdr + 1, req->send.buffer,
```

Comment:
we should rename this to ucp_tag_pack_eager_sync_**single**_dt() and assert the returned length is == req->send.length